include/votca/tools/graphalgorithm.h

```diff
@@ -162,6 +162,52 @@ std::vector<Graph> decoupleIsolatedSubGraphs(Graph graph);
  */
 void exploreGraph(Graph& graph, GraphVisitor& graph_visitor);
 
+/**
+ * @brief Determines how the graph nodes of a particular graph must be arranged
+ * so that no matter what the sequence of nodes is if they are from a
+ * particular graph they are always returned in the same sequence
+ *
+ * This method allows one to find how a graph should be arranged in a canonical
+ * form.
+ *
+ * E.g. Given two of the same graphs but with different ordering of nodes
+ *
+ * Graph A       Graph B
+ * 1a - 2a - 3a      4a - 3a - 2a
+ *       |   |             |   |
+ *      5b - 4b           1b - 5b
+ *
+ * If each graph was passed in, assuming graph A reprsented the canonized form
+ * the sequence would show some thing like this.
+ *
+ * Graph A                     Graph B
+ * 1a(3) - 2a(1) - 3a(4)      4a(3) - 3a(1) - 2a(4)
+ *            |     |                   |       |
+ *         5b(2) - 4b(5)              1b(2) - 5b(5)
+ *
+ * Such that Graph A and B would be returned as vectors, parenthesis indicate
+ * cannonized order.
+ *
+ * Graph A      Graph B
+ * [0] = 2      [0] = 3
+ * [1] = 5      [1] = 1
+ * [2] = 1      [2] = 4
+ * [3] = 3      [3] = 2
+ * [4] = 4      [4] = 5
+ *
+ * These sequences will allow the graph to be reordered such graph A and Graph
+ * B can be mapped onto each other. The algorithm will also return the string
+ * id of the graph in the return value.
+ *
+ * WARNING This algorithm requires that the nodes have been assigned ids.
+ *
+ * @param graph - graph to be examined
+ * @param sequence - vector of ints that contain the ids of the nodes
+ *
+ * @return
+ */
+std::string findCanonizedSequence(Graph& graph, std::vector<Index>& sequence);
```

Comment:
why are the graph and the sequence not `const`? What are the side effects, is there maybe a better name or multiple functions?

Reply:
if sequence is a return value, just return a std::pair<std::string,std::vector>?